Upstream-Status: Backport

Fixes the X11 build, which was failing due to a missing include.

This is sort of a backport: this patch was submitted upstream in
https://webrtc-review.googlesource.com/c/src/+/112282, but was superseded by a
much larger one fixing all includes. Adding just the required change seemed
wiser though.

Signed-off-by: Raphael Kubo da Costa <raphael.kubo.da.costa@intel.com>
---
From fd816463bd2624de984854d6248b05e71bfce761 Mon Sep 17 00:00:00 2001
From: Raphael Kubo da Costa <raphael.kubo.da.costa@intel.com>
Date: Wed, 28 Nov 2018 14:04:49 +0100
Subject: [PATCH] IWYU: Include <string.h> for memset()
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

This fixes the libstdc++ build after 318da51f ('Reland "Add support for
screen sharing with PipeWire on Wayland"'):

    ../../third_party/webrtc/modules/desktop_capture/linux/window_capturer_x11.cc: In member function ‘virtual bool webrtc::WindowCapturerX11::FocusOnSelectedSource()’:
    ../../third_party/webrtc/modules/desktop_capture/linux/window_capturer_x11.cc:122:5: error: ‘memset’ was not declared in this scope
         memset(xev.xclient.data.l, 0, sizeof(xev.xclient.data.l));
         ^~~~~~
    ../../third_party/webrtc/modules/desktop_capture/linux/window_capturer_x11.cc:122:5: note: ‘memset’ is defined in header ‘<cstring>’; did you forget to ‘#include <cstring>’?
    ../../third_party/webrtc/modules/desktop_capture/linux/window_capturer_x11.cc:29:1:
    +#include <cstring>
    ../../third_party/webrtc/modules/desktop_capture/linux/window_capturer_x11.cc:122:5:
         memset(xev.xclient.data.l, 0, sizeof(xev.xclient.data.l));
         ^~~~~~

Bug: chromium:819294, chromium:682122
Change-Id: Icd481f7dc2cfb6f47d7b3127a9431384c92ff480
---
 modules/desktop_capture/linux/window_capturer_x11.cc | 1 +
 1 file changed, 1 insertion(+)

diff --git a/modules/desktop_capture/linux/window_capturer_x11.cc b/modules/desktop_capture/linux/window_capturer_x11.cc
index 4ef7c5cb38..79602c498d 100644
--- a/third_party/webrtc/modules/desktop_capture/linux/window_capturer_x11.cc
+++ b/third_party/webrtc/modules/desktop_capture/linux/window_capturer_x11.cc
@@ -13,6 +13,7 @@
 #include <X11/Xutil.h>
 #include <X11/extensions/Xcomposite.h>
 #include <X11/extensions/Xrender.h>
+#include <string.h>
 
 #include <memory>
 #include <string>
-- 
2.20.1

